src/unisparks/button.cpp

```diff
@@ -336,52 +341,46 @@ void doButtons(Player& player, const Milliseconds currentMillis) {
   uint8_t btn = buttonStatus(0, currentMillis);
 #if BUTTON_LOCK
   // info("doButtons start");
+  // 0 Locked and awaiting click; 1 Awaiting long press; 2 Awaiting click; 3 Awaiting long press; 4 Awaiting release; 5 Unlocked
   static uint8_t buttonLockState = 0;
-  static uint32_t lastButtonTime = 0;
+  static Milliseconds lockButtonTime = 0;
 
-  if (buttonLockState != 0 && currentMillis - lastButtonTime > lockDelay) {
+  // If idle-time expired, return to ‘locked’ state
+  if (buttonLockState != 0 && currentMillis - lockButtonTime >= 0) {
     buttonLockState = 0;
   }
-  if (btn == BTN_RELEASED ||  btn == BTN_LONGPRESS) {
-    lastButtonTime = currentMillis;
-
-    if (buttonLockState == 0) {
-      buttonLockState++;
-    } else if (buttonLockState == 1) {
-      if (btn == BTN_LONGPRESS) {
-        buttonLockState++;
-      } else {
+  if (buttonLockState < 5) {
+    if (BTN_EVENT(btn)) {
+      if (btn != ((buttonLockState & 1) ? BTN_LONGPRESS : BTN_RELEASED)) {
```

Comment:
for consistency with line 369:
```suggestion
      if (btn != ((buttonLockState % 2) ? BTN_LONGPRESS : BTN_RELEASED)) {
```

Reply:
Fixed.